.github/workflows/eden_rpi.yml

```diff
@@ -0,0 +1,134 @@
+          sudo apt install -y curl golang net-tools qemu binfmt-support qemu-user-static qemu-utils qemu-system-x86 qemu-system-aarch64 openvpn jq
+      - name: Setup OVPN connection to RoL RPi4 demo
+        id: vpn
+        run: |
+          echo "$ROL_OVPN_CONF_BASE64" | base64 -d > ./config.ovpn
+          sudo openvpn --config ./config.ovpn --daemon
+          VPN_ATTEMTS_COUNT=0
+          VPN_MAX_ATTEMTS=24
+          until [ $VPN_ATTEMTS_COUNT -eq $VPN_MAX_ATTEMTS ] || ip -f inet addr show tun0; do sleep 5; ip a; let VPN_ATTEMTS_COUNT=VPN_ATTEMTS_COUNT+1; done
+          [ $VPN_ATTEMTS_COUNT -lt $VPN_MAX_ATTEMTS ] || $(echo "::error::VPN connection not established" && exit 1)
+          echo ::set-output name=ip::$(ip -f inet addr show tun0 | sed -En -e 's/.*inet ([0-9.]+).*/\1/p')
+        env:
+          ROL_OVPN_CONF_BASE64: ${{ secrets.ROL_OVPN_CONF_BASE64 }}
+      - name: Prepare eden
+        run: |
+          docker run -v $PWD:/out lfedge/eden:0.4.0 cp -a /eden/. /out/
+          sudo chown -R $(whoami) .
+      - name: Setup eden
+        run: |
+          ./eden config add default --devmodel=general --arch=arm64
+          ./eden config set default --key eve.tag --value="snapshot"
+          ./eden config set default --key eve.hv --value ${{ matrix.hv }}
+          ./eden config set default --key adam.eve-ip --value ${{ steps.vpn.outputs.ip }}
+          ./eden config set default --key registry.ip --value ${{ steps.vpn.outputs.ip }}
+          ./eden config set default --key=eden.tests --value=${{ github.workspace }}/eve/tests/eden
+          ./eden setup -v debug --netboot=true
+          ./eden start
+      - name: Rent RPi4
+        id: rol
+        run: |
+          rent_id=$(./eden rol rent create -p "$ROL_PROJECT" -m raspberry --model pi_4_model_b_8gb -n GHAction-snapshot-${{ matrix.hv }})
+          echo ::set-output name=id::$(echo $rent_id)
+        env:
+          ROL_API_URL: ${{ secrets.ROL_API_URL }}
+          ROL_API_KEY: ${{ secrets.ROL_API_KEY }}
+          ROL_PROJECT: ${{ secrets.ROL_PROJECT }}
+      - name: Waiting for EVE OS to load
+        run: |
+          EVE_ATTEMTS_COUNT=0
+          EVE_MAX_ATTEMTS=360
+          until [ $EVE_ATTEMTS_COUNT -eq $EVE_MAX_ATTEMTS ] || [[ "$(./eden rol rent get -p "$ROL_PROJECT" -i ${{ steps.rol.outputs.id }} | jq -r .machineState)" == "Ready" ]]; do sleep 10; echo "Waiting for EVE to load"; let EVE_ATTEMTS_COUNT=EVE_ATTEMTS_COUNT+1; done
+          [ $EVE_ATTEMTS_COUNT -lt $EVE_MAX_ATTEMTS ] || $(echo "::error::EVE OS boot timeout" && exit 1)
+        env:
+          ROL_API_URL: ${{ secrets.ROL_API_URL }}
+          ROL_API_KEY: ${{ secrets.ROL_API_KEY }}
+          ROL_PROJECT: ${{ secrets.ROL_PROJECT }}
+      - name: EVE Onboard
+        run: |
+          ./eden eve onboard
+      - name: Run eden workflow test
+        run: |
+            EDEN_TEST_REGISTRY=n ./eden test ${{ github.workspace }}/eve/tests/eden/workflow -v debug
```

Comment:
Do we need to skip registry tests? They should work in private networks (VPN).

Reply:
Fixed